### PR TITLE
FEAT/ Verify pool exists before add liquidation 

### PR DESCRIPTION
**Pull request by luanpontolio**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

File: apps/web/src/hooks/useContract.ts [hunk 1]
```diff
@@ -322,7 +322,7 @@ export function useContract<T extends Contract = Contract>(
 }
 
 export function useTokenContract(tokenAddress?: string, abi?: any[], withSignerIfPossible?: boolean) {
-  return useContract<Erc20 | Erc721collection | WrappedNft>(tokenAddress, abi || ERC20_ABI, withSignerIfPossible)
+  return useContract<Erc20 | Erc721collection>(tokenAddress, abi || ERC20_ABI, withSignerIfPossible)
```
Comment (arthurmr96) [inline apps/web/src/hooks/useContract.ts:325]: Is it not better to change the ERC20 ABI for the Wrapped NFT? To make available the collection property also when the desired address is a WNFT

**Review by arthurmr96 — CHANGES_REQUESTED** [review]

File: apps/web/src/hooks/usePairs.ts [hunk 2]
```diff
@@ -32,7 +34,7 @@ export function usePairs(currencies: [Currency | undefined, Currency | undefined
     () =>
       tokens.map(([tokenA, tokenB]) => {
         try {
-          return tokenA && tokenB && !tokenA.equals(tokenB) ? Pair.getAddress(tokenA, tokenB) : undefined
+          return tokenA && tokenB && !tokenA.equals(tokenB) ? computePairAddress({ factoryAddress: FACTORY_ADDRESS[chainId], tokenA, tokenB }) : undefined
```
Comment (arthurmr96) [inline apps/web/src/hooks/usePairs.ts:37]: `computePairAddress` does the same thing the `getAddress` did but with the correct factory?

File: apps/web/src/state/user/hooks/useUserAddedTokens.ts [hunk 3]
```diff
@@ -12,7 +12,7 @@ export const userAddedTokenSelector = (chainId: number) =>
   createSelector(selectUserTokens, (serializedTokensMap) =>
     Object.values(serializedTokensMap?.[chainId] ?? {}).map(deserializeToken),
   )
-export default function useUserAddedTokens(): Token[] {
+export default function useUserAddedTokens(): Token[] | WrappedNftErc20Token[] {
```
Comment (arthurmr96) [inline apps/web/src/state/user/hooks/useUserAddedTokens.ts:15]:
```suggestion
export default function useUserAddedTokens(): WrappedNftErc20Token[] {
```

Comment (arthurmr96) [reply apps/web/src/state/user/hooks/useUserAddedTokens.ts:15]: `WrappedNftErc20Token` already extends `Token`

File: apps/web/src/state/wallet/hooks.ts [hunk 4]
```diff
@@ -46,14 +48,14 @@ export function useNativeBalances(uncheckedAddresses?: (string | undefined)[]):
...
 export function useTokenBalancesWithLoadingIndicator(
   address?: string,
-  tokens?: (Token | undefined)[],
-): [{ [tokenAddress: string]: CurrencyAmount<Token> | undefined }, boolean] {
-  const validatedTokens: Token[] = useMemo(
-    () => tokens?.filter((t?: Token): t is Token => isAddress(t?.address) !== false) ?? [],
+  tokens?: (Token | WrappedNftErc20Token | undefined)[],
+): [{ [tokenAddress: string]: CurrencyAmount<Token | WrappedNftErc20Token> | undefined }, boolean] {
+  const validatedTokens: Token[] | WrappedNftErc20Token[] = useMemo(
+    () => tokens?.filter((t?: Token | WrappedNftErc20Token): t is Token | WrappedNftErc20Token => isAddress(('collection' in t && t?.collection) || t?.address) !== false) ?? [],
```
Comment (arthurmr96) [inline apps/web/src/state/wallet/hooks.ts:54]: `WrappedNftErc20Token` already extends `Token`

**Review by arthurmr96 — CHANGES_REQUESTED** [review]

File: apps/web/src/hooks/usePairs.ts (hunk 2 above)
Comment (luanpontolio) [reply apps/web/src/hooks/usePairs.ts:37]: The `Pair.getAddress` is a method used by pancake-swap to find the LP token for the pair. 
In the SDK the wrong address was used for `FACTORY` and `INIT_CODE_HASH.`

File: apps/web/src/hooks/useContract.ts (hunk 1 above)
Comment (luanpontolio) [reply apps/web/src/hooks/useContract.ts:325]: Ok. Maybe we can change WrappedNft to Erc20Wrapped. What do you think?

File: apps/web/src/hooks/usePairs.ts (hunk 2 above)
Comment (arthurmr96) [reply apps/web/src/hooks/usePairs.ts:37]: but all that it was being done on the method, is being doing by the new method with the correct addresses?

**Review by arthurmr96 — APPROVED** [review]

**Review by arthurmr96 — APPROVED** [review]

**Pull request merged**